### PR TITLE
S-014a: restrict ReadSecrets to service principals, and redact uniformly

### PR DESCRIPTION
**Pull request by benhegartysefe**
Closes #849

**Stacked on #848.** Merge that first. This is the last PR in the stack.

## ⚠️ Blast radius — worth a heads-up before this merges

**Every environment owner who today sees decrypted secret values in the web UI will stop seeing them.** Anyone relying on that to debug a deployment or confirm a credential rotation landed loses that view. Worth finding out who relies on it before shipping, not after.

**Windows authentication carries no machine-versus-person discriminator**, so `IsServicePrincipal` is false for every Windows-authenticated caller and the privilege cannot be exercised over that scheme at all. That is the intended shape — service accounts authenticate as OAuth machine clients, which is where the `m2m` claim exists — but a Windows-auth consumer that reads secrets would stop working. The Monitor is unaffected: it uses the direct-tool reader, which reports itself a machine.

## What changes

`CanReadSecrets` requires a service principal **and** an explicit grant. Ownership no longer implies the privilege, so an audit of holders means something. The `m2m` determination moves from a private method in the OAuth claims reader onto `IClaimsPrincipalReader`, so the authorization layer can ask the question.

## The predicate separation, which the plan required a decision on

The controller's second use of the predicate is **not** ACL visibility — it is the guard that stops a caller granting `ReadSecrets` when they do not hold it. Left on the restricted predicate, **no human could ever have granted the privilege to anyone**, and it would have become unadministrable the moment this shipped.

So there are now two predicates: `CanReadSecrets` for exercising it, and `CanGrantReadSecrets` for administering it, keeping the original privilege-or-ownership semantics. Administration behaves exactly as before.

## Redaction corrections

Humans are redacted rather than refused, per the agreed usage decision. Both edge cases in the issue are fixed, because this step turns them from what a few callers hit into what every human hits.

## Tests

6 new; one existing test updated from asserting the throw to asserting uniform redaction.

---
_Generated by [Claude Code](https://claude.ai/code/session_01KT2gJnb8LC4DGHT2bDTCJk)_

### CODE REVIEW

File: src/Dorc.Api/Controllers/AccessControlController.cs [hunk 1]
```diff
@@ -129,34 +129,53 @@ public IActionResult Put(AccessSecureApiModel accessControl)
...
+                foreach (var privilege in accessControl.Privileges.Where(p => p.Id > 0))
                 {
-                    var existingPrivileges = _accessControlPersistentSource.GetAccessControls(accessControl.ObjectId)
-                        .ToDictionary(p => p.Id);
+                    if (!existingPrivileges.ContainsKey(privilege.Id))
+                    {
+                        return StatusCode(StatusCodes.Status400BadRequest,
+                            "An access-control entry does not belong to the authorized object.");
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (hegsie) [reply src/Dorc.Api/Controllers/AccessControlController.cs]: Already addressed in the current branch: the permission sequence is filtered with `.Where(p => p.Id > 0)` before iteration.

**Review by kirill-suspitsin-ext-sefe — APPROVED** [review]
